pio/lib/Sender/Sender.cpp

```diff
@@ -11,10 +11,140 @@
 #include <ThingSpeak.h>
 #include <BlynkSimpleEsp8266.h> //https://github.com/blynkkk/blynk-library
 
+// For AWS IoT
+#include <WiFiClientSecure.h>
+#include <ArduinoJson.h>
+#include <MQTT.h>
+#include <time.h>
+
+#define emptyString String()
+
+#include "../../secrets.h"
+
+#if !(ARDUINOJSON_VERSION_MAJOR == 6 and ARDUINOJSON_VERSION_MINOR >= 7)
+#error "Install ArduinoJson v6.7.0-beta or higher"
+#endif
+
+const int MQTT_PORT = 8883;
+char topic[100];
+char jsonBuffer[200];
+String response;
+
+const char MQTT_TOPIC_PFX[] = "$aws/things/";
+const char MQTT_PUB_TOPIC_SFX[] = "/shadow/update";
+const char MQTT_SUB_TOPIC_SFX[] = "/shadow/update/delta";
+// const char MQTT_SUB_TOPIC[] = "$aws/things/" THINGNAME "/shadow/update";
+// const char MQTT_PUB_TOPIC[] = "$aws/things/" THINGNAME "/shadow/update";
+
+#ifdef USE_SUMMER_TIME_DST
+uint8_t DST = 1;
+#else
+uint8_t DST = 0;
+#endif
+
+WiFiClientSecure net;
+
+#ifdef ESP8266
+BearSSL::X509List cert(cacert);
+BearSSL::X509List client_crt(client_cert);
+BearSSL::PrivateKey key(privkey);
+#endif
+
+MQTTClient _awsIoTClient(200);
+
+unsigned long lastMillis = 0;
+time_t now;
+time_t nowish = 1510592825;
+
+// End of for AWS IoT
+
 #define UBISERVER "things.ubidots.com"
 #define BLYNKSERVER "blynk-cloud.com"
 #define CONNTIMEOUT 2000
 
+void NTPConnect(void)
```

Comment:
The battery life concept is "fire and forget" as every second runtime significantly reduces battery life time. Therefor I avoid lookups by all means which eventually cause delays or potentially even hang.

Reply:
Yes, in TLS, verifying certificate validity is optional, and since we normally settle for regular http, I guess security is not as much an issue, as it is something to "get past".

I´ll have a look at scrapping time lookup, I have actually had a hangup during a brew recently, switched to plain old http after that.